### ISSUE TITLE
Development instructions?

### ISSUE THREAD

**Issue by marckohlbrugge** [cou929/stability-sdk-ruby#13]
Would be great if the README included development instructions for the gem.

Stability's API is moving fast (e.g. they added CLIP guidance recently) and we can't really expect you to immediately update the moment a new version comes out.

However, it would be nice if in the mean time we can easily build our temporary fork with the latest API changes incorporated. I'm not exactly sure how though.

I did find https://github.com/kmcphillips/stability-sdk/blob/main/src/ruby/README.md – what parts of this should I run to update my local copy of the gem to have the latest proto stuff?

**Comment by cou929** [comment]
I agree. I think that it's useful if README contains such development instructions.

As a short answer, I write down how to build the client with the latest protobuf.
I haven't actually tried this out. So it might be needed to try and error. I will update README after I can confirm that this is correct.

1. update the submodule (stability-ai/api-interfaces) to the latest

    ```sh
    cd api-interfaces
    git checkout main
    git pull
    cd ..
    ```

2. build with protoc

    ```sh
    bundle exec rake protoc
    
    git diff
    # now you may be able to confirm that the diff is created in lib/generation_pb.rb
    ```

3. modify the client.rb to use new interfaces
    - You can add some lines to use new request parameters in [Client#generate](https://github.com/cou929/stability-sdk-ruby/blob/d2cf7efe18cb46b9d97b8588a9774a2caa3c7691/lib/stability_sdk/client.rb#L50-L92)
    - In the case of CLIP guidance, referring to the implementation of an official python client might be helpful
        - https://github.com/Stability-AI/stability-sdk/blob/f9e2aca5b2a3462333abae2e4cb06dacb9d7193c/src/stability_sdk/client.py#L313-L338

**Comment by marckohlbrugge** [comment]
Thanks for this!

I followed the instructions, but `./api-interfaces` remained empty.

I first had to change the submodule URL to use http instead of ssh, which might be due to my personal config.

```sh
git submodule set-url api-interfaces https://github.com/Stability-AI/api-interfaces.git
```

Afterwards I could run this:

```sh
git submodule update --recursive --init
```

And finally I followed your instructions (along with `bundle install`).

I'm now at step 3 which is just regular Ruby code. Which I should be able to figure out 👍

**Comment by cou929** [comment]
I've added an instruction https://github.com/cou929/stability-sdk-ruby#update-submodule-to-try-a-new-feature-on-stability-aiapi-interfaces

Also supported CLIP guidance in https://github.com/cou929/stability-sdk-ruby/pull/16 and released as 0.2.12
